namer/dnssrv/src/main/scala/io/buoyant/namer/dnssrv/DnsSrvNamer.scala

```diff
@@ -0,0 +1,76 @@
+package io.buoyant.namer.dnssrv
+
+import java.io.IOException
+import java.net.InetSocketAddress
+
+import com.twitter.finagle._
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.logging.Logger
+import com.twitter.util.Activity.State
+import com.twitter.util._
+import org.xbill.DNS
+
+class DnsSrvNamer(prefix: Path, resolver: DNS.Resolver, timer: Timer, refreshInterval: Duration, stats: StatsReceiver)
+  extends Namer {
+
+  override def lookup(path: Path): Activity[NameTree[Name]] = memoizedLookup(path)
+
+  private val log = Logger.get("dnssrv")
+  private val memoizedLookup: (Path) => Activity[NameTree[Name]] = Memoize { path =>
+    path.take(1) match {
+      case id@Path.Utf8(address) =>
+        Activity(Var.async[State[NameTree[Name]]](Activity.Pending) { state =>
+          timer.schedule(refreshInterval) {
+            val next = lookupSrv(address, prefix ++ id, path.drop(1)) match {
+              case Return(nameTree) => Activity.Ok(nameTree)
+              case Throw(e) => Activity.Failed(e)
+            }
+            state.update(next)
+          }
```

Comment:
just to confirm i understand this... timer.schedule returns a closable and the timer is canceleld when the Var observation is released?

Reply:
yes, that is correct